webapp/src/js/services/live-list.js

```diff
@@ -51,8 +51,15 @@ angular.module('inboxServices').factory('LiveListConfig',
           if (c1.simprints && c2.simprints) {
             return c2.simprints.confidence - c1.simprints.confidence;
           }
-          if (c1.type !== c2.type) {
-            return ContactSchema.getTypes().indexOf(c1.type) - ContactSchema.getTypes().indexOf(c2.type);
+          const c1Type = c1.contact_type || c1.type;
+          const c2Type = c2.contact_type || c2.type;
+          if (c1Type !== c2Type) {
+            const c1TypeIndex = ContactTypes.HARDCODED_TYPES.indexOf(c1Type);
+            const c2TypeIndex = ContactTypes.HARDCODED_TYPES.indexOf(c2Type);
+            if (c1TypeIndex !== c2TypeIndex) {
+              return c1TypeIndex - c2TypeIndex;
+            }
+            return c1Type - c2Type;
```

Comment:
This piece of code might end up sorting the contacts differently than the `contacts_by_type` view (which is the view called when sorting contacts alphabetically). So, when we have pagination, subsequent pages end up pushing contacts to the top like so:
Before loading all pages:
![image](https://user-images.githubusercontent.com/35681649/58039072-5b348900-7b3a-11e9-994b-47b035de9d9d.png)
After loading all pages:
![image](https://user-images.githubusercontent.com/35681649/58039107-756e6700-7b3a-11e9-9483-c7ff324f4576.png)

As a bit of reference, I only added a new place type called "area" (contact_types.id = area) which ends up getting compared with "clinic" and "health_center", while in the view, it ends up being compared with an array index and is at the end of the view results list.

Reply:
Right. I've changed the sorting logic to match the view. Thanks!

The sorting may be a little strange when comparing hardcoded types which sort based on their index, and configured types which sort on their name, eg: comparing "3" to "area". But it should now be consistent at least...